lib/database/repositories/LogRepository.js

```diff
@@ -21,17 +21,41 @@ const { timestampToMysql } = require('../../server/utilities/timestampToMysql.js
  * @param {Period} the period in which logs should have been created
  * @return {string} the query
  */
-const getTagDistributionQuery = ({ from, to }) => `
+const getTagDistributionQuery = ({ from, to }) => sequelize.query(`
     SELECT t.text tag, COUNT(*) count
     FROM logs l
              INNER JOIN log_tags lt on l.id = lt.log_id
              INNER JOIN tags t on lt.tag_id = t.id
-    WHERE l.created_at >= '${timestampToMysql(from)}'
-      AND l.created_at < '${timestampToMysql(to)}'
+    WHERE l.created_at >= :fromTimeStamp
+      AND l.created_at < :toTimeStamp
       AND t.archived_at IS NULL
     GROUP BY t.text
     ORDER BY COUNT(*) DESC, t.text
-`;
+`, {
+    replacements: { fromTimeStamp: timestampToMysql(from), toTimeStamp: timestampToMysql(to) },
+});
+
+/**
+ * Returns the query to fetch a log's reply count for a given parent log id.
+ *
+ * @param {number} parentLogId The parent log id for which the reply count will be fetced.
+ * @returns {Promise<Array<{ reply_count: number }>>} A promise resolving to an array of objects,
+ * each containing the reply count for the given parent log id.
+ */
+const getLogsWithReplyCountQuery = (parentLogId) => sequelize.query(`
+    WITH RECURSIVE reply_tree AS (
+        SELECT id, parent_log_id
+        FROM logs
+        WHERE parent_log_id = :parentLogId
+        UNION ALL
+        SELECT logs.id, logs.parent_log_id
+        FROM logs
+        JOIN reply_tree ON logs.parent_log_id = reply_tree.id
+    )
+    SELECT COUNT(*) AS reply_count FROM reply_tree;
+`, {
+    replacements: { parentLogId },
+});
```

Comment:
The idea is here, nice job.
The main issue I see here is that you do one query per log entry, which is quite expensive if you have a lot of logs.
I propose to use instead:
```SQL
WITH RECURSIVE log_tree (originLogId, id) AS (
    SELECT id originId, id FROM logs -- In the anchor query, store the root log for which we want the tree
    WHERE id IN (7687, 7808) -- Pick the logs we are interested in, all at once
    UNION ALL
    SELECT originLogId, l.id FROM logs as l -- Then recurse, but always keep the origin to be able to group on it
        JOIN log_tree AS lt ON lt.id = l.parent_log_id
)
SELECT COUNT(*) - 1 FROM log_tree GROUP BY originLogId;
```